### ISSUE TITLE
Demangle _ZN4glslL7combineIhLi2EEEDvmlT0_Li4E_T_DvT0__S1_S3_S3_S3_

### ISSUE THREAD

**Issue by mstange** [gimli-rs/cpp_demangle#231]
I don't know if any other demangler supports this, but it was produced by llvm and observed [in this profile](https://share.firefox.dev/3EkXOLr):

```
_ZN4glslL7combineIhLi2EEEDvmlT0_Li4E_T_DvT0__S1_S3_S3_S3_
```

cpp_demangle currently demangles it to

```
glsl::combine<unsigned char, 2>DvmlT0_Li4E_T_DvT0__S1_S3_S3_S3_
```

I think the correct demangling would be:

```
glsl::combine<unsigned char, 2>(unsigned char __vector(2), unsigned char __vector(2), unsigned char __vector(2), unsigned char __vector(2))
```

This function is [defined here](https://searchfox.org/mozilla-central/rev/3d03a3ca09f03f06ef46a511446537563f62a0c6/gfx/wr/swgl/src/vector_type.h#414-416) and called with `T = uint8_t` and `N = 2`:

```cpp
template <typename T, int N>
SI VectorType<T, N * 4> combine(VectorType<T, N> a, VectorType<T, N> b,
                                VectorType<T, N> c, VectorType<T, N> d) {
```

**Comment by khuey** [comment]
Yeah we recognize Dv as the start of a vector type but then it's followed by ml instead of a number or a type substitution and we bail out.

**Comment by khuey** [comment]
The symbol `_ZN4glslL7combineIhLi2EEEDv_mlT0_Li4E_T_Dv_T0__S1_S3_S3_S3_` demangles with cpp_demangle to `unsigned char __vector((2)*(4)) glsl::combine<unsigned char, 2>(unsigned char __vector(2), unsigned char __vector(2), unsigned char __vector(2), unsigned char __vector(2))` which is more or less what you expect. So the issue here is that LLVM is not emitting the expected (in libiberty) `_` before the dimension expression.

**Comment by mstange** [comment]
Thank you!

**Comment by khuey** [comment]
I've decided libiberty is wrong (or rather, not used in the wild) and changed cpp_demangle to follow LLVM.

**Comment by khuey** [comment]
And I've published 0.3.5 with this change.
